integration-tests/tests/build.gradle

```diff
@@ -11,3 +13,31 @@ dependencies {
 
   kaptTest deps.dagger2.compiler
 }
+
+// Okay, this check is sketchy here. But this module serves as integration test so it's not too
+// terrible. What would be worse is setting up integration tests in the Gradle plugin itself,
+// because it requires so much code and might not represent the real world usage.
+//
+// Kotlin 1.3 is broken. The hacky workarounds don't always work. The issue seems to be gone with
+// Kotlin 1.4, so enable this check only for the newer version.
+
+// TODO: remove this check when upgrading to Kotlin 1.4.
+def enableCheck = VersionNumber.parse(rootProject.ext.kotlinVersion).baseVersion >= VersionNumber.parse("1.4.10")
+if (enableCheck) {
+  gradle.taskGraph.afterTask { Task task ->
+    if (task.project != project) return
+
+    task.inputs.properties.each { key, value ->
+      if (key.contains('com.squareup.anvil.compiler.src-gen-dir')) {
```

Comment:
Is this something that should be checked by using `contains` or `==`? Is the full input string much longer/dynamic?

Reply:
Definitely contains. The Kotlin Gradle plugin adds `kotlinCompile.$key` as property for some tasks. It's really weird.